orc/src/main/java/org/apache/iceberg/orc/ColumnMap.java

```diff
@@ -33,11 +35,48 @@
  * Keep the API limited to Map rather than a concrete type so that we can
  * change it later.
  */
-public class ColumnIdMap implements Map<TypeDescription, Integer> {
+public final class ColumnMap implements Map<TypeDescription, ColumnMap.IcebergColumn> {
```

Comment:
Extending map seems to require a lot of extra boilerplate code that delegates to the internal map.

I think it would be cleaner to make serialize and inverse static methods that accept any `Map<TypeDescription, IcebergColumn>`.